### ISSUE TITLE
Workgroup size should never be larger than the capacity of the group

### ISSUE THREAD

**Issue by pcwalton** [djeedai/bevy_hanabi#389]
We hard-code a workgroup size of 64, even if the group has capacity 1. This will always make 63 of the GPU threads go to waste. We should cap the workgroup size at the capacity.

**Comment by djeedai** [comment]
Good point. On the other hand, the hardware will never spawn less than its internal workgroup size limit, which is at least 32 for some vendors, and 64 for others. So the threads will go to waste either way. Having a hard-coded limit simplifies a tiny bit some places (hard-coded conversions, instead of having to inject a size).

**Comment by djeedai** [comment]
Closing this as this actually adds complexity without providing any benefit. The hardware will never spawn 1 thread, it always spawns units of 32 or 64 anyway, and any other value is a combination of those. So we're not wasting anything.
